hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/AbstractGenericRecord.java

```diff
@@ -87,35 +87,35 @@ public int hashCode() {
     private static int arrayHashCode(GenericRecord record, String path, FieldType type) {
         switch (type) {
             case BYTE_ARRAY:
-                return Arrays.hashCode(record.readByteArray(path));
+                return Arrays.hashCode(record.getByteArray(path));
             case SHORT_ARRAY:
-                return Arrays.hashCode(record.readShortArray(path));
+                return Arrays.hashCode(record.getShortArray(path));
             case INT_ARRAY:
-                return Arrays.hashCode(record.readIntArray(path));
+                return Arrays.hashCode(record.getIntArray(path));
             case LONG_ARRAY:
-                return Arrays.hashCode(record.readLongArray(path));
+                return Arrays.hashCode(record.getLongArray(path));
             case FLOAT_ARRAY:
-                return Arrays.hashCode(record.readFloatArray(path));
+                return Arrays.hashCode(record.getFloatArray(path));
             case DOUBLE_ARRAY:
-                return Arrays.hashCode(record.readDoubleArray(path));
+                return Arrays.hashCode(record.getDoubleArray(path));
             case BOOLEAN_ARRAY:
-                return Arrays.hashCode(record.readBooleanArray(path));
+                return Arrays.hashCode(record.getBooleanArray(path));
             case CHAR_ARRAY:
-                return Arrays.hashCode(record.readCharArray(path));
+                return Arrays.hashCode(record.getCharArray(path));
             case UTF_ARRAY:
```

Comment:
As mentioned on the pr comments, I could not change this because of backward compatibility. 
This is a public API and part of a enum. Especially the following method is problematic:
```
public static FieldType get(byte type);
``` 

This was returning `UTF` and `UTF_ARRAY` so far. If I change it, it means we can potentially break the codes of relying on the string representations of these two. So, I decided not to do any change here.